functions/Get-DbaDbccMemoryStatus.ps1

```diff
@@ -0,0 +1,106 @@
+
+        Reference:
+            - https://blogs.msdn.microsoft.com/timchapman/2012/08/16/how-to-parse-dbcc-memorystatus-via-powershell/
+            - https://support.microsoft.com/en-us/help/907877/how-to-use-the-dbcc-memorystatus-command-to-monitor-memory-usage-on-sq
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances.
+
+    .PARAMETER SqlCredential
+        Login to the target instance using alternative credentials. Windows and SQL Authentication supported. Accepts credential objects (Get-Credential)
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: DBCC, Memory
+        Author: Patrick Flynn (@sqllensman)
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaMemoryStatus
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbccMemoryStatus -SqlInstance sqlcluster, sqlserver2012
+
+        Get output of DBCC MEMORYSTATUS for instances "sqlcluster" and "sqlserver2012". Returns results in a single recordset.
+
+    .EXAMPLE
+        PS C:\> Get-DbaCmsRegServer -SqlInstance sqlcluster | Get-DbaDbccMemoryStatus
+
+        Get output of DBCC MEMORYSTATUS for all servers in Server Central Management Server
+
+          #>
+    [CmdletBinding()]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [switch]$EnableException
+    )
+    begin {
+        $query = 'DBCC MEMORYSTATUS'
+    }
+    process {
+        foreach ($instance in $SqlInstance) {
+            Write-Message -Message "Attempting Connection to $instance" -Level Verbose
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            Write-Message -Level Verbose -Message "Collecting $query data from server: $instance"
+            try {
+                $sqlconnection = New-Object System.Data.SqlClient.SqlConnection
+                $sqlconnection.ConnectionString = $server.ConnectionContext.ConnectionString
+                $sqlconnection.Open()
+
+                $datatable = New-Object system.Data.DataSet
+                $dataadapter = New-Object system.Data.SqlClient.SqlDataAdapter($query, $sqlconnection)
```

Comment:
Can this be `$server.Query()` instead?

Reply:
Query does not seem to return the multiple resultsets thay are needed to return information for this command. If there is an option to dp this then happy to revert to standard method.

Reply:
makes sense, no need, thank you 👍